wasi/wasi.go

```diff
@@ -386,3 +400,9 @@ func environ_sizes_get(*wasm.HostFunctionCallContext, uint32, uint32) (err Errno
 func environ_get(*wasm.HostFunctionCallContext, uint32, uint32) (err Errno) {
 	return
 }
+
+func clock_time_get(ctx *wasm.HostFunctionCallContext, id uint32, precision uint64, timestampPtr uint32) (err Errno) {
+	nanos := uint64(time.Now().UnixNano())
```

Comment:
rather than directly using `time.Now().UnixNano()`, how about having `clockTimeFunc func() uint64)` as a field of WASIEnvironment and default it to `time.Now().UnixNano` for testability?

Reply:
Love it

Reply:
Done